### PR TITLE
MEN-7818: Add explicit test updating from standalone state v1

### PR DESCRIPTION
**Pull request by lluiscampos**
See bug fix at:
* https://github.com/mendersoftware/mender/pull/1705

### CODE REVIEW

File: tests/test_update.py [hunk 1]
```diff
@@ -1171,3 +1172,41 @@ def test_rollback_after_commit(
...
+                )
+
+            # Make sure the database is initialized by reading it with show-provides
+            connection.run(f"{mender_update_binary} show-provides")
+
+            # Update the database on the device with the v1 standalone state data
+            database_path = Path(tmpdir, "mender-store")
+            get_no_sftp("/var/lib/mender/mender-store", connection, database_path)
+            subprocess.check_call(
+                ["/usr/bin/mdb_load", "-T", "-f", state_v1_path, "-n", database_path]
```
Comment (vpodzime) [inline tests/test_update.py:1203]: I didn't know `mdb_load` supported JSON. Cool!

File: tests/test_update.py [hunk 2]
```diff
@@ -1171,3 +1172,41 @@ def test_rollback_after_commit(
...
+            )
+            put_no_sftp(database_path, connection, "/var/lib/mender/mender-store")
+
+            # Create the working directory for the "ongoing" update
+            connection.run("mkdir -p /var/lib/mender/modules/v3/payloads/0000/tree")
+
+            # The database indicates an ongoing update from v1. Commit it
+            output = connection.run(f"{mender_update_binary} -l trace commit")
+            assert output.stdout.rstrip("\n") == "Committed."
```
Comment (vpodzime) [inline tests/test_update.py:1212]: Should we check the new state of the DB?

**Review by vpodzime — APPROVED** [review]
Looks good to me either way.

File: tests/test_update.py (hunk 1 above)
Comment (lluiscampos) [reply tests/test_update.py:1203]: It does not support JSON per se, it supports a key/value pair of strings (two lines), and our value just happens to be JSON formatted.

File: tests/test_update.py (hunk 2 above)
Comment (lluiscampos) [reply tests/test_update.py:1212]: We could, but we would only see that the state does not exist anymore. Regardless of the update being committed  or rolled back the entry in the database is removed.

However, you reminding me that what I wanted is to add a  `show-provides` call and get the new artifact name! I'll do that :+1:  (or maybe that is what you were referring to?)

Comment (lluiscampos) [reply tests/test_update.py:1212]: Done

**Review by danielskinstad — APPROVED** [review]

File: tests/test_update.py (hunk 1 above)
Comment (vpodzime) [reply tests/test_update.py:1203]: oh! Now I see it!

File: tests/test_update.py (hunk 2 above)
Comment (vpodzime) [reply tests/test_update.py:1212]: > or maybe that is what you were referring to?

Exactly :grin: I just thought it would be in the DB.

**Comment by lluiscampos** [comment]
Tested in https://github.com/mendersoftware/meta-mender/pull/2223 :green_circle:

**Pull request merged**